src/txn/txn_rollback_to_stable.c

```diff
@@ -251,12 +251,19 @@ __rollback_row_ondisk_fixup_key(WT_SESSION_IMPL *session, WT_PAGE *page, WT_ROW
         WT_ERR(hs_cursor->get_value(
           hs_cursor, &hs_stop_durable_ts, &hs_durable_ts, &type_full, hs_value));
         type = (uint8_t)type_full;
-        if (type == WT_UPDATE_MODIFY)
-            WT_ERR(__wt_modify_apply_item(
-              session, S2BT(session)->value_format, &full_value, hs_value->data));
-        else {
-            WT_ASSERT(session, type == WT_UPDATE_STANDARD);
-            WT_ERR(__wt_buf_set(session, &full_value, hs_value->data, hs_value->size));
+
+        /*
+         * Do not include history store updates greater than on-disk data store version to construct
+         * a full update to restore.
+         */
+        if (hs_start_ts <= unpack->tw.start_ts) {
```

Comment:
Can you add a comment why it is required and ok to check timestamp in rts but not ok in search?

Reply:
Added.